storage/src/main/java/org/apache/kafka/server/log/remote/metadata/storage/ConsumerManager.java

```diff
@@ -76,47 +94,50 @@ public void startConsumerThread() {
     }
 
     /**
-     * Waits if necessary for the consumption to reach the offset of the given {@code recordMetadata}.
+     * Waits if necessary for the consumption to reach the {@code offset} of the given record
+     * at a certain {@code partition} of the metadata topic.
      *
-     * @param recordMetadata record metadata to be checked for consumption.
+     * @param partition partition of metadata topic to be checked for consumption
+     * @param offset record offset of metadata topic {@code partition} to be checked for consumption
      * @throws TimeoutException if this method execution did not complete with in the wait time configured with
      *                          property {@code TopicBasedRemoteLogMetadataManagerConfig#REMOTE_LOG_METADATA_CONSUME_WAIT_MS_PROP}.
      */
-    public void waitTillConsumptionCatchesUp(RecordMetadata recordMetadata) throws TimeoutException {
-        waitTillConsumptionCatchesUp(recordMetadata, rlmmConfig.consumeWaitMs());
+    public void waitTillConsumptionCatchesUp(int partition, long offset) throws TimeoutException {
+        waitTillConsumptionCatchesUp(partition, offset, rlmmConfig.consumeWaitMs());
     }
 
     /**
-     * Waits if necessary for the consumption to reach the offset of the given {@code recordMetadata}.
+     * Waits if necessary for the consumption to reach the {@code offset} of the given record
+     * at a certain {@code partition} of the metadata topic.
      *
-     * @param recordMetadata record metadata to be checked for consumption.
-     * @param timeoutMs      wait timeout in milli seconds
+     * @param partition partition of metadata topic to be checked for consumption
+     * @param offset record offset of metadata topic {@code partition} to be checked for consumption
+     * @param timeoutMs wait timeout in milliseconds
      * @throws TimeoutException if this method execution did not complete with in the given {@code timeoutMs}.
      */
-    public void waitTillConsumptionCatchesUp(RecordMetadata recordMetadata,
-                                             long timeoutMs) throws TimeoutException {
-        final int partition = recordMetadata.partition();
+    public void waitTillConsumptionCatchesUp(int partition, long offset, long timeoutMs) throws TimeoutException {
         final long consumeCheckIntervalMs = Math.min(CONSUME_RECHECK_INTERVAL_MS, timeoutMs);
```

Comment:
I think we should log something as INFO when waiting the consumer catching up, ex:
`log.info("waiting until consumer is caught up with the target partition {}", partition);`
so that we can know why it's pending without enable debug log. WDYT?

Reply:
This makes sense especially since we are dealing with remote storage and hence we may get high lag times.